### ISSUE TITLE
Design System Focus Styles

### ISSUE THREAD

**Issue by BrittanyIRL** [GoogleForCreators/web-stories-wp#6099]
## Feature Description

After some back and forth we're going to change the way that focus is approached in the design system. 
Focus styles are only present for keyboard users. 
Focus style should be a "2px space" and then a 2px border focus color. I put space in quotes because it's not actually a space, this is going to be a background color. The color of the "space" should default to the variant's bg.primary but it should be able to be overridden. The focus blue will always be the same.  https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3438%3A125786

This focus style should be sharable so that we have one single source of truth. It should only show for keyboard users. It should not affect the position or spacing of elements. 

The fallback focus outline in the design system should also be updated to be the same as the one the editor uses.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

- Sharable focus style is updated in design system
- Default focus outline updated in design system to match editor 
- All design system components that use a focus style updated to use shareable focus style.

**Comment by BrittanyIRL** [comment]
New focus styles can be seen on staging storybook here: https://google.github.io/web-stories-wp/storybook/?path=/story/designsystem-components-button--default
